packages/preferences/src/node/json-preference-server.ts

```diff
@@ -100,27 +101,39 @@ export class JsonPreferenceServer implements PreferenceServer {
             if (this.preferences) {
                 this.fireChanged(this.preferences, preferences);
             } else {
-                this.fireNew(preferences);
+                // Will not fire individual events until the server has finished reading the preferences once
+                if (this.isServerReady) {
+                    this.fireNew(preferences);
+                }
             }
         } else if (this.preferences) {
             this.fireRemoved(this.preferences);
         }
         this.preferences = preferences;
+        // Send an event containing all preferences once
+        if (!this.isServerReady) {
+            this.fireInitEvent();
+        }
+        this.isServerReady = true;
+    }
+
+    protected fireInitEvent(): void {
+        this.fireEvent({ preferenceName: "", newValue: this.preferences });
     }
 
     protected fireNew(preferences: any): void {
         // tslint:disable-next-line:forin
         for (const preferenceName in preferences) {
             const newValue = preferences[preferenceName];
-            this.fireEvent({ preferenceName, newValue });
+            this.fireEvent({ preferenceName: preferenceName, newValue: newValue });
         }
     }
 
     protected fireRemoved(preferences: any): void {
         // tslint:disable-next-line:forin
         for (const preferenceName in preferences) {
             const oldValue = preferences[preferenceName];
-            this.fireEvent({ preferenceName, oldValue });
+            this.fireEvent({ preferenceName: preferenceName, oldValue: oldValue });
```

Comment:
I wasn't sure, if the property isn't specified, doesn't the value sent as argument go to the next optional property (i.e newValue in this case).